celo-org/celo-monorepo
# FIX - Support forno endpoint on make-release script.

### Description

Adds new `-F` flag to `make-release.sh` to support for the `--forno` flag on the truffle config.

This helps as a workaround on a config error when trying to run `yarn make-release ...` with a custom RPC endpoint setup.

### Other changes
NA

### Tested

1. Replicate the issue.
With forno endpoint set as an env var:
```shell
export CELO_NODE_ADDRESS="https://alfajores-forno.celo-testnet.org"

 yarn make-release  -b $RELEASE_CANDIDATE -n $NETWORK -r "report.json" -i "releaseData/initializationData/release6.json" -p "proposal.json" -l "libraries.json" -d

yarn run v1.22.10
$ ./scripts/bash/make-release.sh -b core-contracts.v6 -n alfajores -r report.json -i releaseData/initializationData/release6.json -p proposal.json -l libraries.json -d
 - Checkout contracts source code at core-contracts.v6
 - Contract artifacts already built at build/core-contracts.v6
$ /Users/luisoscargabriel/luisgj/celo-monorepo/node_modules/.bin/truffle exec ./scripts/truffle/make-release.js --network alfajores --build_directory build/core-contracts.v6 --report report.json --librariesFile libraries.json --proposal proposal.json --from --initialize_data releaseData/initializationData/release6.json --dry_run
> Warning: possible unsupported (undocumented in help) command line option: --build_directory,--report,--librariesFile,--proposal,--from,--initialize_data,--dry_run
> Something went wrong while attempting to connect to the network. Check your network configuration.

Could not connect to your Ethereum client with the following parameters:
    - host       > https://alfajores-forno.celo-testnet.org
    - port       > 8545
    - network_id > 44787
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle-config.js)

Truffle v5.1.18 (core: 5.1.18)
Node v14.5.0
error Command failed with exit code 1.
```

2. Test the fix by seting up `--forno` flag
```shell
yarn make-release  -b $RELEASE_CANDIDATE -n $NETWORK -r "report.json" -i "releaseData/initializationData/release6.json" -p "proposal.json" -l "libraries.json" -d -F


yarn run v1.22.10
$ ./scripts/bash/make-release.sh -b core-contracts.v6 -n alfajores -r report.json -i releaseData/initializationData/release6.json -p proposal.json -l libraries.json -d -F
 - Checkout contracts source code at core-contracts.v6
 - Contract artifacts already built at build/core-contracts.v6
$ /Users/luisoscargabriel/luisgj/celo-monorepo/node_modules/.bin/truffle exec ./scripts/truffle/make-release.js --network alfajores --build_directory build/core-contracts.v6 --report report.json --librariesFile libraries.json --proposal proposal.json --from --initialize_data releaseData/initializationData/release6.json --dry_run --forno
> Warning: possible unsupported (undocumented in help) command line option: --build_directory,--report,--librariesFile,--proposal,--from,--initialize_data,--dry_run,--forno
Using network 'alfajores'.

Deploying ExchangeBRL
Deploying ExchangeBRLProxy
Transferring ownership of ExchangeBRLProxy to Governance
.
.
.
```

### Backwards compatibility

This just adds an optional flag. The script behaves the same.

### Documentation

_The set of community facing docs that have been added/modified because of this change_
